protos/sources/data-node/api/v2/trading_data.proto

```diff
@@ -4780,3 +4790,36 @@ message PartyMarginMode {
   // Epoch at which the update happened.
   uint64 at_epoch = 7;
 }
+
+// Time weighted notional position for a party and asset at a given epoch. It is used as a metric to calculate whether
+// the party is eligible for receiving a reward.
+message TimeWeightedNotionalPosition {
+  // Asset ID for the asset associated with the position.
+  string asset_id = 1;
+  // Party ID for the party associated with the position.
+  string party_id = 2;
+  // Epoch at which the time weighted notional position is calculated.
+  uint64 at_epoch = 3;
+  // Time weighted notional position for the party and asset.
+  string time_weighted_notional_position = 4;
+  // Vega time of the block in which the last position update occurred.
+  int64 last_updated = 5;
+}
+
+// Request to retrieve the time weighted notional position for a party and asset.
+// If no epoch is given, the most current value will be returned.
```

Comment:
Depending on changes to functionality, could you update this (and other instances) if it becomes “last epoch” rather than live current?